lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1101 @@
+        LoadDataFromPDB(PDBFileNameOnly.str());
+      } else if (SearchPathPDB.size() > 0) {
+        // Otherwise, try to find the PDB in user defined paths.
+        for (const std::string &Path : SearchPathPDB) {
+          std::string UserDefinedPDBFilePath = Path + PDBFileNameOnly.str();
+          if (llvm::sys::fs::exists(UserDefinedPDBFilePath)) {
+            LoadDataFromPDB(UserDefinedPDBFilePath);
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  if (not ThePDBFile) {
+    revng_log(DILogger, "Unable to find PDB file.");
+    return;
+  } else {
+    // According to llvm/docs/PDB/PdbStream.rst, the `Signature` was never
+    // used the way as it was the initial idea. Instead, GUID is a 128-bit
+    // identifier guaranteed to be unique ID for both executable and
+    // corresponding PDB.
+    codeview::GUID GUIDFromExe;
+    llvm::copy(DebugInfo->PDB70.Signature, std::begin(GUIDFromExe.Guid));
+    auto PDBInfoStrm = ThePDBFile->getPDBInfoStream();
+    if (!PDBInfoStrm)
+      revng_log(DILogger, "No PDB Info stream found.");
+    else {
+      codeview::GUID GUIDFromPDBFIle = PDBInfoStrm->getGuid();
+      if (GUIDFromExe != GUIDFromPDBFIle)
+        revng_log(DILogger, "Signatures from exe and PDB file mismatch.");
+    }
+  }
+
+  PDBImporterImpl ModelCreator(*this);
+  ModelCreator.run(*TheNativeSession);
+}
+
+// ==== Implementation of the Model type recordings. ==== //
+
+Error PDBImporterTypeVisitor::visitTypeBegin(CVType &Record) {
+  return visitTypeBegin(Record, TypeIndex::fromArrayIndex(Types.size()));
+}
+
+Error PDBImporterTypeVisitor::visitTypeBegin(CVType &Record, TypeIndex TI) {
+  CurrentTypeIndex = TI;
+  return Error::success();
+}
+
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               FieldListRecord &FieldList) {
+  if (auto EC = visitMemberRecordStream(FieldList.Data, *this))
+    return EC;
+  return Error::success();
+}
+
+// Determine the pointer size based on CodeView/PDB data.
+static uint32_t getPointerSizeInBytes(codeview::PointerKind K) {
+  switch (K) {
+  case codeview::PointerKind::Near64:
+    return 8;
+  case codeview::PointerKind::Near32:
+    return 4;
+  default:
+    // TODO: Handle all pointer kinds.
+    revng_abort();
+  }
+}
+
+// Parse LF_POINTER.
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               PointerRecord &Ptr) {
+  using namespace model;
+  auto TypeTypedef = makeType<TypedefType>();
+
+  TypeIndex ReferencedType = Ptr.getReferentType();
+  if (ProcessedTypes.count(ReferencedType) == 0 and ReferencedType.isSimple())
+    createPrimitiveType(ReferencedType);
+
+  if (ProcessedTypes.count(ReferencedType)) {
+    auto PointerSize = getPointerSizeInBytes(Ptr.getPointerKind());
+    std::vector<Qualifier> Qualifiers{ Qualifier::createPointer(PointerSize) };
+    QualifiedType TheUnderlyingType(ProcessedTypes[ReferencedType], Qualifiers);
```

Comment:
Concerning the lazy creation of primitive types: having to check each time `if (ProcessedTypes.count(ReferencedType) == 0 and ReferencedType.isSimple())` is a bit fragile. It's easy to forget.

Can you make a method that wraps fetching the `ProcessedTypes[ReferencedType]` so that it first performs a look up and then, in case of failure, automatically tries to materialize the `PrimitiveType` (if possible, aborts otherwise).